yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -549,6 +549,21 @@ private[spark] class Client(
       appMasterOnly = true)
     require(confLocalizedPath != null)
 
+    // Uploading $SPARK_CONF_DIR/log4j.properties file to the distributed cache to make sure that
+    // the executors will use the latest configurations instead of the default values. This is
+    // required when user changes log4j.properties directly to set the log configurations. If
+    // configuration file is provided through --files then executors will be taking configurations
+    // from --files instead of $SPARK_CONF_DIR/log4j.properties.
+
+    // Also uploading metrics.properties to distributed cache if exists in classpath.
+    // If user specify this file using --files then executors will use the one
+    // from --files instead.
+    for { prop <- Seq("log4j.properties", "metrics.properties")
```

Comment:
Since its deprecated and I would like to see it removed I don't think its that big of deal, but I disagree with the order if we are keeping it.

If I explicitly specify something in SPARK_LOG4J_CONF it should take precendence over anything in the <SPARK_HOME>/conf dir.